lib/DxilPIXPasses/DxilDebugInstrumentation.cpp

```diff
@@ -455,13 +458,13 @@ DxilDebugInstrumentation::addRequiredSystemValues(BuilderContext &BC,
     // about the shader having selected components that don't include x or y.
     // If not present, we add it.
     if (Existing_SV_Position == InputElements.end()) {
+      unsigned int Index = static_cast<unsigned int>(InputElements.size());
       auto Added_SV_Position =
           llvm::make_unique<DxilSignatureElement>(DXIL::SigPointKind::PSIn);
       Added_SV_Position->Initialize("Position", hlsl::CompType::getF32(),
-                                    hlsl::DXIL::InterpolationMode::Linear, 1,
-                                    4);
+                                    hlsl::DXIL::InterpolationMode::Linear, 1, 4,
+                                    Index, 0);
```

Comment:
I think there are potential problems with this:

Index is being passed to the `StartRow` parameter, which is supposed to the be the packing location starting row, not the index of the element.  This row must match the upstream vertex shader's output row for `SV_Position`, whatever that is.

You should not be relying on default parameter values for `ID` and `IndexVector` when constructing an element for final DXIL.  These initial states are meant for an earlier phase, before element flattening, index assignment, and signature packing.

Reply:
By the way, `ID` and `IndexVector` are easy to supply, they are the index of the element in the signature (`InputElements.size()`) and a single-element vector containing the single semantic index for your 1-row element, which for each of your use cases is `0`.

Reply:
I think this latest change does it... I have some PIX-side tests that do the round-trip from the new input sig reader pass back to the debug instrumentation pass.